vta/hardware/chisel/src/main/scala/core/Configs.scala

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package vta.core
+
+import vta.util.config._
+
+class CoreConfig extends Config((site, here, up) => {
+  case CoreKey => CoreParams(
```

Comment:
VTA configurations won't be the usual `config` file that you can modify and expect for the best. Configurations in Chisel will be using this style see [1](https://github.com/freechipsproject/rocket-chip/blob/master/src/main/scala/system/Configs.scala) and [2](https://github.com/freechipsproject/rocket-chip/blob/master/src/main/scala/subsystem/Configs.scala). Only supported configurations will be available in those files, rather than user modifying that. This is because Chisel is imperative, you have to add support explicitly, and not declarative compared to HLS.

Most of the modules are "parametrizable" but there are some stuff that currently need some thought, specially regarding the IO (host and memory) interaction.

Reply:
so these are the default values that will be overwritten when the module is instantiated?